astropy/nddata/nddata_base.py

```diff
@@ -90,3 +90,9 @@ def uncertainty(self, value):
                 raise TypeError('Uncertainty must have attribute '
                                 'uncertainty_type whose type is string.')
         self._uncertainty = value
+        #Try setting the parent for the uncertainty. This is currently an
+        #enforced requirement for `StdDevUncertainty` for error propagation.
+        try:
+            self.uncertainty.parent_nddata = self
+        except:
```

Comment:
That breaks a lot of existing test cases since we allow the uncertainty to be ``None`` which has no (parent_nddata attribute) but this could be fixed if I shift some lines around.

But thats unfortunatly not all: The ``test_nddata_base`` defines as minimal uncertainty something that has no ``parent_nddata`` attribute (https://github.com/astropy/astropy/blob/master/astropy/nddata/tests/test_nddata_base.py#L36). That made wonder what is considered the Minimal Uncertainty I should enforce here and I finally ended up with a ``try ... except pass`` which adds no restriction.